### DISCUSSION TITLE
Problems with attach database

### DISCUSSION THREAD

**Discussion by wlodek** [pawelsalawa/sqlitestudio#4542] [Q&A]
### Details
I have been using SQLiteStudio for a long time, but for a few weeks I have had a problem with attaching database. I receive error message "Error while executing SQL query on database 'dbname': file is not a database". But the file is a valid database - I work with it and it works. I tried with other valida database - but the result is always the same.

### Steps to reproduce
Connect with the database SQL CIPHER using PRAGMA CIPHER_COMPATIBILITY = 3;
Go to editor, select opened database and try to:
ATTACH DATABASE xxx AS ccc KEY ddd;
where xxx is also SQL CIPHER
I had never problems with this, until now.

### Operating system
WIndows 11, Mac OS Ventura (also previous version of Mac OS)

### SQLiteStudio version
3.3.3

**Comment by pawelsalawa** [comment]
Are you able to open that problematic (the `xxx`) database directly? (I mean by adding it to the list in SQLiteStudio and opening it as primary database, not attached)

**Comment by wlodek** [comment]
Yes

niedz., 6 lis 2022, 21:27 użytkownik Paweł Salawa ***@***.***>
napisał:

**Comment by wlodek** [comment]
Again - yes, but using pragma cipher_compatibility=3

niedz., 6 lis 2022, 21:27 użytkownik Paweł Salawa ***@***.***>
napisał:

**Comment by pawelsalawa** [comment]
SQLiteStudio is just using SQLCipher here and if attaching with it is not working, there is not much SQLiteStudio can do about it.

According to official SQLCipher docs (https://www.zetetic.net/sqlcipher/sqlcipher-api/#attach) it's not possible to define compatibility levels when attaching. That may be the root cause of your issue.

You can try to ask SQLCipher support for help on their forums: https://discuss.zetetic.net/c/sqlcipher/5
